src/Nethermind/Nethermind.Serialization.Rlp/TxDecoder.cs

```diff
@@ -2,887 +2,164 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
 using Microsoft.Extensions.ObjectPool;
 using Nethermind.Core;
-using Nethermind.Core.Crypto;
-using Nethermind.Core.Extensions;
-using Nethermind.Serialization.Rlp.Eip2930;
+using Nethermind.Serialization.Rlp.MyTxDecoder;
 
-namespace Nethermind.Serialization.Rlp
+namespace Nethermind.Serialization.Rlp;
+
+public sealed class TxDecoder : TxDecoder<Transaction>
 {
-    public sealed class TxDecoder : TxDecoder<Transaction>
-    {
-        public const int MaxDelayedHashTxnSize = 32768;
-        public static TxDecoder Instance = new TxDecoder();
-        public static TxDecoder InstanceWithoutLazyHash = new TxDecoder(false);
-        public static ObjectPool<Transaction> TxObjectPool = new DefaultObjectPool<Transaction>(new Transaction.PoolPolicy(), Environment.ProcessorCount * 4);
+    public static readonly ObjectPool<Transaction> TxObjectPool = new DefaultObjectPool<Transaction>(new Transaction.PoolPolicy(), Environment.ProcessorCount * 4);
 
-        public TxDecoder() : base(true) // Rlp will try to find empty constructor.
-        {
-        }
+#pragma warning disable CS0618
+    public static readonly TxDecoder Instance = new();
+#pragma warning restore CS0618
 
-        public TxDecoder(bool lazyHash) : base(lazyHash)
-        {
-        }
-
-        protected override Transaction NewTx()
-        {
-            return TxObjectPool.Get();
-        }
-    }
-    public class SystemTxDecoder : TxDecoder<SystemTransaction> { }
-    public class GeneratedTxDecoder : TxDecoder<GeneratedTransaction> { }
+    // Rlp will try to find a public parameterless constructor during static initialization.
+    // The lambda cannot be removed due to static block initialization order.
+    [Obsolete("Use `TxDecoder.Instance` instead")]
+    public TxDecoder() : base(() => TxObjectPool.Get()) { }
```

Comment:
C# does not provide a good attribute to mark a method as "do not use" and the common solution for this case is to use `Obsolete` (see: https://stackoverflow.com/questions/27758/notify-developer-of-a-do-not-use-method)